src/main/scala/se/lth/immun/DinoUtil.scala

```diff
@@ -299,4 +299,10 @@ object DinoUtil {
 		val k = (x-x0) / (x1-x0)
 		k*y1 + (1-k)*y0
 	}
-}
\ No newline at end of file
+	
+	def getMinMaxIndx(sortedArray:Array[Double], x:Double, tol:Double):(Int, Int) = {
+  	val startIndx = math.abs(java.util.Arrays.binarySearch(sortedArray, x - tol)+1)
```

Comment:
I'm not entirely sure this does what you want if `x - tol` is present in `sortedArray`. I guess this code base doesn't do tests much (my fault), but a unit test would have been helpful here.

Reply:
If I understand the documentation right, if `x - tol` is present in `sortedArray` it will return the index of that element. Is your suggestion to remove the `+1`?